### PR TITLE
Add global.cattle.psp.enabled to rancher-monitoring charts

### PR DESCRIPTION
**Pull request by aiyengar2**
This will only cover the main Monitoring chart and all subcharts. Project Monitoring is done in a separate PR to pull in from upstream.

Related Issue: https://github.com/rancher/rancher/issues/40350, https://github.com/rancher/rancher/issues/40372

### CODE REVIEW

File: charts/rancher-monitoring/102.0.0+up40.1.2/charts/grafana/templates/role.yaml [hunk 1]
```diff
@@ -10,9 +10,9 @@ metadata:
   annotations:
 {{ toYaml . | indent 4 }}
 {{- end }}
-{{- if or .Values.rbac.pspEnabled (and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled (or .Values.sidecar.datasources.enabled (or .Values.sidecar.plugins.enabled .Values.rbac.extraRoleRules)))) }}
+{{- if or .Values.global.cattle.psp.enabled (and .Values.rbac.namespaced (or .Values.sidecar.dashboards.enabled (or .Values.sidecar.datasources.enabled (or .Values.sidecar.plugins.enabled .Values.rbac.extraRoleRules)))) }}
```
Comment (stormqueen1990) [inline charts/rancher-monitoring/102.0.0+up40.1.2/charts/grafana/templates/role.yaml:13]: Would we need a `validate-psp-install.yaml` in this subchart too?

File: charts/rancher-monitoring/102.0.0+up40.1.2/charts/rke2Etcd/templates/pushprox-proxy-rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  name: {{ template "pushProxy.proxy.name" . }}
+  labels: {{ include "pushProxy.proxy.labels" . | nindent 4 }}
+rules:
+{{- if .Values.global.cattle.psp.enabled }}
+- apiGroups: ['policy']
+  resources: ['podsecuritypolicies']
+  verbs:     ['use']
+  resourceNames:
+  - {{ template "pushProxy.proxy.name" . }}
+  {{- end }}
```
Comment (stormqueen1990) [inline charts/rancher-monitoring/102.0.0+up40.1.2/charts/rke2Etcd/templates/pushprox-proxy-rbac.yaml]: nit: do we want this condition to have `if` and `end` aligned with one another?

File: charts/rancher-pushprox/102.0.0/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+annotations:
+  catalog.cattle.io/hidden: "true"
+  catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.25.0-0'
```
Comment (stormqueen1990) [inline charts/rancher-pushprox/102.0.0/Chart.yaml]: I think this needs to be updated.

File: index.yaml [hunk 4]
```diff
@@ -10930,6 +10930,24 @@ entries:
     - assets/rancher-prometheus-adapter/rancher-prometheus-adapter-2.12.101.tgz
     version: 2.12.101
   rancher-pushprox:
+  - annotations:
+      catalog.cattle.io/hidden: "true"
+      catalog.cattle.io/kube-version: '>= 1.16.0-0 < 1.25.0-0'
```
Comment (stormqueen1990) [inline index.yaml]: I guess this gets updated when `Chart.yaml` gets updated too?

**Review by stormqueen1990 — COMMENTED** [review]
This LGTM, I just have a few questions.

File: charts/rancher-monitoring/102.0.0+up40.1.2/charts/grafana/templates/role.yaml (hunk 1 above)
Comment (aiyengar2) [reply charts/rancher-monitoring/102.0.0+up40.1.2/charts/grafana/templates/role.yaml:13]: No need. We decided that only standalone charts need it; so rancher-pushprox and rancher-monitoring only.

File: charts/rancher-monitoring/102.0.0+up40.1.2/charts/rke2Etcd/templates/pushprox-proxy-rbac.yaml (hunk 2 above)
Comment (aiyengar2) [reply charts/rancher-monitoring/102.0.0+up40.1.2/charts/rke2Etcd/templates/pushprox-proxy-rbac.yaml]: good catch. Fixing.

File: index.yaml (hunk 4 above)
Comment (aiyengar2) [reply index.yaml]: yes

File: charts/rancher-pushprox/102.0.0/Chart.yaml (hunk 3 above)
Comment (aiyengar2) [reply charts/rancher-pushprox/102.0.0/Chart.yaml]: Good catch. Will update in this and main chart.

**Review by stormqueen1990 — APPROVED** [review]
LGTM!

**Pull request merged**